Content.Server/Lathe/LatheSystem.cs

```diff
@@ -54,217 +57,172 @@ public override void Update(float frameTime)
                 RemCompDeferred(comp.Owner, comp);
             }
 
-            foreach (var comp in EntityQuery<LatheProducingComponent>())
+            foreach (var comp in EntityQuery<LatheComponent>())
+            {
+                if (!comp.Queue.Any() || !this.IsPowered(comp.Owner, EntityManager))
+                    continue;
+
+                TryStartProducing(comp.Owner, comp);
+            }
+
+            foreach (var (comp, lathe) in EntityQuery<LatheProducingComponent, LatheComponent>())
             {
                 comp.TimeRemaining -= frameTime;
 
                 if (comp.TimeRemaining <= 0)
-                    FinishProducing(comp.Owner, comp);
+                    FinishProducing(comp.Owner, lathe);
             }
```

Comment:
Wouldn't it be better to add LatheProducingComponent if the queue ever goes > 0 then check powered in the bottom method?